taxonomy-editor-frontend/src/pages/editentry/AccumulateAllComponents.jsx

```diff
@@ -14,46 +15,104 @@ import ListAllEntryRelationships from "./ListAllEntryRelationships";
  * If node is "header/footer": Comments are rendered  
 */ 
 const AccumulateAllComponents = ({ id }) => {
+
+    // Finding URL to send requests
     const url = createURL(id);
     const isEntry = getIdType(id) === 'entry';
-    const [nodeObject, setNodeObject] = useState(null); // Storing node information
+
+    const [nodeObject, setNodeObject] = useState(null); // Storing original node information
+    const [updatedNodeObject, setUpdatedNodeObject] = useState(null); // Storing updates to node
+    const [updateChildren, setUpdateChildren] = useState([]); // Storing updates of children in node
+    const [open, setOpen] = useState(false); // Used for Dialog component
+    const navigate = useNavigate(); // Navigation between pages
     const { data: node, isPending, isError, isSuccess, errorMessage } = useFetch(url);
 
     // Setting state of node after fetch
     useEffect(() => {
         setNodeObject(node?.[0]);
```

Comment:
does this needs to be in a state? what about using directly as `const originalNode = node?.[0]`

Reply:
Yep, it needs to be in a state. Since if assignment happens inside the useEffect, that variable's scope isn't global. If it was defined in a 'let', the contents will be lost after each render.

Reply:
Yep, changed it to a derived state.